src/WorldStorage/FastNBT.cpp

```diff
@@ -503,7 +503,7 @@ void cFastNBTWriter::AddString(const AString & a_Name, const AString & a_Value)
 void cFastNBTWriter::AddByteArray(const AString & a_Name, const char * a_Value, size_t a_NumElements)
 {
 	TagCommon(a_Name, TAG_ByteArray);
-	u_long len = htonl(static_cast<u_long>(a_NumElements));
+	u_int len = htonl(static_cast<u_int>(a_NumElements));
```

Comment:
`htonl` consumes `u_long` and returns `u_long`, at least as declared in Windows SDK.